v2/api/authorization/v1api20220401/role_assignment_defaults.go

```diff
@@ -43,13 +45,18 @@ func (assignment *RoleAssignment) defaultAzureName() {
 	}
 
 	if assignment.AzureName() == "" {
-		gk := assignment.GroupVersionKind().GroupKind()
-		assignment.Spec.AzureName = randextensions.MakeUUIDName(
-			assignment.Name,
-			randextensions.MakeUniqueOwnerScopedStringLegacy(
-				assignment.Owner(),
-				gk,
-				assignment.Namespace,
-				assignment.Name))
+		if assignment.Spec.OperatorSpec != nil &&
+			strings.ToLower(*assignment.Spec.OperatorSpec.UUIDGeneration) == "random" {
+			assignment.Spec.AzureName = randextensions.MakeRandomUUID()
+		} else {
```

Comment:
If the user supplied value isn't one of the enum, we won't assign `AzureName` and we'll get a natural error.

Also, update the doc-comment above to reflect the new behaviour.

Reply:
Wouldn't that be a breaking change? 
Thought would be good if we do the default behavior if nothing is mentioned? 

Also, I'd update `deterministic` to be `default`

Reply:
or I think `stable` and `random` would be good too. WDYT?

Reply:
My concern is that if someone set the property to `Jamaican` we would accept that and do something with it! I really don't think we should.

I'd retain the existing behavour if someone leaves it blank (`""`) _or_ requests it explicitly (`"stable"`), and the new behaviour if someone sets it to `"random"`.

I don't really favour `default` because it's not clear what that means, other than it's the default. I like `stable` though, that's good.

Reply:
We have the validations to take care of that case if the user sets a disallowed value. I think that should take care of that?